pymc3/distributions/dist_math.py

```diff
@@ -381,18 +382,17 @@ def perform(self, node, inputs, output_storage):
         x, = inputs
         output_storage[0][0] = np.asarray(self.spline(x))
 
-class DifferentiableSplineWrapper (SplineWrapper):
-    """
-    Creates a theano operation with defined gradient from
-    scipy.interpolate.UnivariateSpline
-    """
-
-    def __init__(self, spline):
-        super(DifferentiableSplineWrapper, self).__init__(spline)
-        self.spline_grad = SplineWrapper(spline.derivative())
-        self.__props__ += ('spline_grad',)
-
     def grad(self, inputs, grads):
         x, = inputs
         x_grad, = grads
-        return [x_grad * self.spline_grad(x)]
+
+        if not hasattr(self, 'grad_op'):
+            try:
+                self.grad_op = SplineWrapper(self.spline.derivative())
```

Comment:
I mean what would possible usages? This class is anyway internal to PyMC3, so I think that we need in it only the functionality that is actually used by PyMC3.